### PR TITLE
Giant monster refactor

### PR DESCRIPTION
**Pull request by BeeHiveJava**
-Refactored **A LOT** of classes.
-Documented **A LOT** of classes.

Do note: the specified documentation isn't actually using the conventions, bounds still need to be declared etc.

### CODE REVIEW

File: Soccar [Client]/src/nl/soccar/ui/fx/PhysicsDrawableFx.java [hunk 1]
```diff
@@ -3,18 +3,34 @@
 import nl.soccar.ui.physics.WorldObject;
 
 /**
- *
+ * This class is an extension on the regular Fx Drawable. On top of the regular implementation
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/PhysicsDrawableFx.java]: On top of the regular implementation**.**

File: Soccar [Client]/src/nl/soccar/ui/fx/PhysicsDrawableFx.java [hunk 2]
```diff
@@ -3,18 +3,34 @@
...
  * @author PTS34A
+ * @param <M> The model that is connected to this Drawable.
+ * @param <P> The physics-model that is connected to this Drawable.
  */
 public abstract class PhysicsDrawableFx<M, P extends WorldObject> extends DrawableFx<M> {
 
     private P physicsModel;
 
+    /**
+     * Initiates a new PhysicsDrawabkeFx
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/PhysicsDrawableFx.java]: Initiates a new PhysicsDrawabkeFx **.**

File: Soccar [Client]/src/nl/soccar/ui/fx/PhysicsDrawableFx.java [hunk 3]
```diff
@@ -3,18 +3,34 @@
...
+     * 
+     * @param canvas The canvas on which this Drawable is placed.
+     * @param model The model of this Drawable.
+     * @param physicsModel The physics-model Drawable.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/PhysicsDrawableFx.java]: The physics-model of this Drawable.

File: Soccar [Client]/src/nl/soccar/ui/fx/models/BallUiFx.java [hunk 4]
```diff
@@ -9,13 +9,23 @@
...
  * @author PTS34A
  */
 public class BallUiFx extends PhysicsDrawableFx<Ball, BallPhysics> {
 
     private static final Color COLOR = Color.BLACK;
 
+    /**
+     * Initializes a new BallUiFx Object using the given parameters.
+     * 
+     * @param canvas The canvas on which this Ball is drawn.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/models/BallUiFx.java]: @param canvas The canvas on which this Ball is **placed**.

File: Soccar [Client]/src/nl/soccar/ui/fx/models/CarUiFx.java [hunk 5]
```diff
@@ -27,6 +30,13 @@
         TEXTURE_CAR_RED = new Image(DisplayConstants.LOCATION_TEXTURE_CAR_RED);
     }
 
+    /**
+     * Initiates a new CarUiFx Object using the given parameters.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/models/CarUiFx.java:34]: **Initializes** a new CarUiFx Object using the given parameters.

File: Soccar [Client]/src/nl/soccar/ui/fx/models/MapUiFx.java [hunk 6]
```diff
@@ -29,6 +31,12 @@
         TEXTURE_GRASS = new Image(DisplayConstants.LOCATION_TEXTURE_GRASS);
     }
 
+    /**
+     * Initiates a new MapUiFx Object using the given parameters.
+     *
+     * @param canvas The canvas on which this Map is drawn.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/models/MapUiFx.java]: @param canvas The canvas on which this Map is **placed**.

File: Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java [hunk 7]
```diff
@@ -9,17 +9,27 @@
 import nl.soccar.util.PhysicsUtilities;
 
 /**
+ * A Obstacle object represents a JavaFX Drawable of an Obstacle. It keeps track
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java]: A **ObstacleUiFx** object represents a JavaFX Drawable of an Obstacle.

File: Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java [hunk 8]
```diff
@@ -9,17 +9,27 @@
...
     private static final Color WALL_COLOR;
-    
+
     static {
         WALL_COLOR = Color.BLACK;
     }
-    
+
+    /**
+     * Initiates a new ObstacleUiFx Object using the given parameters.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java:27]: **Initializes** a new ObstacleUiFx Object using the given parameters.

File: Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java [hunk 9]
```diff
@@ -9,17 +9,27 @@
...
+     *
+     * @param canvas The canvas on which this Obstacle is drawn.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java]: @param canvas The canvas on which this Obstacle is **placed**.

File: Soccar [Client]/src/nl/soccar/ui/physics/WorldObject.java [hunk 10]
```diff
@@ -1,17 +1,42 @@
...
+    /**
+     * Gets the y-coordinate of this physics-model, relative to the map it is 
+     * placed in.
+     * 
+     * @return The y-coordinate of this WorldObject.
+     */
     float getY();
     
+    /**
+     * Gets the angle of this physics-model, relative to the map it is placed
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/physics/WorldObject.java:35]: Angle of Degree???????

File: Soccar [Client]/src/nl/soccar/ui/physics/models/BallPhysics.java [hunk 11]
```diff
@@ -11,18 +11,25 @@
...
-    private final float RESTITUTION = 1.0F;
+    private static final float DENSITY = 1.0F;
+    private static final float FRICTION = 0.0F;
+    private static final float RESTITUTION = 1.0F;
 
     private final Body body;
     private final float radius;
 
+    /**
+     * Initiates a new BallPhysics Object using the given parameter.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/physics/models/BallPhysics.java:28]: **Initializes** a new BallPhysics Object using the given parameter.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/BallPhysics.java [hunk 12]
```diff
@@ -11,18 +11,25 @@
...
+     * 
+     * @param ball The model to keep track of.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/physics/models/BallPhysics.java]: @param ball The **ball** model to keep track of.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/CarPhysics.java [hunk 13]
```diff
@@ -41,6 +42,12 @@
 
     private List<WheelPhysics> wheels;
 
+    /**
+     * Initiates a new CarPhysics Object using the given parameters.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/physics/models/CarPhysics.java:46]: **Initializes** a new CarPhysics Object using the given parameters.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/ObstaclePhysics.java [hunk 14]
```diff
@@ -10,18 +9,26 @@
...
-    private final float FRICTION = 0.0F;
+    private static final float FRICTION = 0.0F;
 
     private final Body body;
 
     private float width;
     private float height;
 
+    /**
+     * Initiates a new ObstaclePhysics Object using the given parameters.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/physics/models/ObstaclePhysics.java:27]: **Initializes** a new ObstaclePhysics Object using the given parameters.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/WheelPhysics.java [hunk 15]
```diff
@@ -31,6 +33,19 @@
 
     private float acceleration = 0;
 
+    /**
+     * Initiates a new WheelPhysics Object using the given parameters.
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/ui/physics/models/WheelPhysics.java:37]: **Initializes** a new WheelPhysics Object using the given parameters.

File: Soccar [Client]/src/nl/soccar/util/PhysicsUtilities.java [hunk 16]
```diff
@@ -4,47 +4,84 @@
...
+     * @return The calculated wheel width.
+     */
     public static float calculateWheelWidth(float carWidth) {
         return carWidth / 5;
     }
-    
-    // Calculate relative wheel height based on hiven car width
+
+    /**
+     * Calculate relative wheel height based on given car width
```
Comment (Ranray) [inline Soccar [Client]/src/nl/soccar/util/PhysicsUtilities.java]: **Calculates** relative wheel height based on given car width

**Review by Ranray — CHANGES_REQUESTED** [review]

File: Soccar [Client]/src/nl/soccar/ui/fx/models/CarUiFx.java (hunk 5 above)
Comment (BeeHiveJava) [reply Soccar [Client]/src/nl/soccar/ui/fx/models/CarUiFx.java:34]: Ik gebruik overal initiates.

File: Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java (hunk 8 above)
Comment (BeeHiveJava) [reply Soccar [Client]/src/nl/soccar/ui/fx/models/ObstacleUiFx.java:27]: Ik gebruik overal Initiates.

File: Soccar [Client]/src/nl/soccar/ui/physics/WorldObject.java (hunk 10 above)
Comment (BeeHiveJava) [reply Soccar [Client]/src/nl/soccar/ui/physics/WorldObject.java:35]: Angle is goed.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/BallPhysics.java (hunk 11 above)
Comment (BeeHiveJava) [reply Soccar [Client]/src/nl/soccar/ui/physics/models/BallPhysics.java:28]: Ik gebruik overal initiates.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/CarPhysics.java (hunk 13 above)
Comment (BeeHiveJava) [reply Soccar [Client]/src/nl/soccar/ui/physics/models/CarPhysics.java:46]: Ik gebruik overal initiates.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/ObstaclePhysics.java (hunk 14 above)
Comment (BeeHiveJava) [reply Soccar [Client]/src/nl/soccar/ui/physics/models/ObstaclePhysics.java:27]: Ik gebruik overal initiates.

File: Soccar [Client]/src/nl/soccar/ui/physics/models/WheelPhysics.java (hunk 15 above)
Comment (BeeHiveJava) [reply Soccar [Client]/src/nl/soccar/ui/physics/models/WheelPhysics.java:37]: Ik gebruik overal initiates.

**Review by Ranray — APPROVED** [review]

**Pull request merged**